rules/kor/kor-rr.js

```diff
@@ -0,0 +1,187 @@
+( function ( $ ) {
+	'use strict';
+
+	var patternList = [
+		// Syllable finals
+		[ '([ᅡ-ᅵ])k', '$1ᆨ' ],
+		[ 'ᆨk', 'ᆩ' ],
+		[ 'ᆨs', 'ᆪ' ],
+		[ '([ᅡ-ᅵ])n', '$1ᆫ' ],
+		[ 'ᆫj', 'ᆬ' ],
+		[ 'ᆫh', 'ᆭ' ],
+		[ '([ᅡ-ᅵ])t', '$1ᆮ' ],
+		[ '([ᅡ-ᅵ])l', '$1ᆯ' ],
+		[ '([ᅡ-ᅵ])r', '$1ᆯ' ],
+		[ 'ᆯk', 'ᆰ' ],
+		[ 'ᆯm', 'ᆱ' ],
+		[ 'ᆯb', 'ᆲ' ],
+		[ 'ᆯs', 'ᆳ' ],
+		[ 'ᆯt', 'ᆴ' ],
+		[ 'ᆯp', 'ᆵ' ],
+		[ 'ᆯh', 'ᆶ' ],
+		[ '([ᅡ-ᅵ])m', '$1ᆷ' ],
+		[ '([ᅡ-ᅵ])b', '$1ᆸ' ],
+		[ 'ᆸs', 'ᆹ' ],
+		[ '([ᅡ-ᅵ])s', '$1ᆺ' ],
+		[ 'ᆺs', 'ᆻ' ],
+		[ 'ᆫg', 'ᆼ' ],
+		[ '([ᅡ-ᅵ])j', '$1ᆽ' ],
+		[ '([ᅡ-ᅵ])ch', '$1ᆾ' ],
+		[ '([ᅡ-ᅵ])K', '$1ᆿ' ],
+		[ '([ᅡ-ᅵ])T', '$1ᇀ' ],
+		[ '([ᅡ-ᅵ])p', '$1ᇁ' ],
+		[ '([ᅡ-ᅵ])h', '$1ᇂ' ],
+
+		// Use space, hyphen, and apostrophe to disambiguate
+		// Do nothing, combineJamo will do the work 
+		[ '([\- \'])', '$1'],
+
+		// Syllable initials
+		[ 'g', 'ᄀ' ],
+		[ 'ᄀk', 'ᄁ' ],
+		[ 'n', 'ᄂ' ],
+		[ 'ᄃt', 'ᄄ' ],
+		[ 'ᄐt', 'ᄄ' ],
+		[ 'ᄃd', 'ᄄ' ],
+		[ 'ᄐd', 'ᄄ' ],
+		[ 'd', 'ᄃ' ],
+		[ 'r', 'ᄅ' ],
+		[ 'l', 'ᄅ' ],
+		[ 'm', 'ᄆ' ],
+		[ 'b', 'ᄇ' ],
+		[ 'ᄇp', 'ᄈ' ],
+		[ 'ᄉs', 'ᄊ' ],
+		// [ '\'', 'ᄋ'],  // Apostrophe can be written to represent silent ᄋ
+		[ 's', 'ᄉ' ],
+		[ 'ᄌj', 'ᄍ' ],
+		[ 'j', 'ᄌ' ],
+		[ 'ch', 'ᄎ' ],
+		[ 'k', 'ᄏ' ],
+		[ 'K', 'ᄏ' ],  // There is some ambiguity for final ᆿ and ᇀ, so they get capital K and T
+		[ 't', 'ᄐ' ],
+		[ 'T', 'ᄐ' ],
+		[ 'p', 'ᄑ' ],
+		[ 'h', 'ᄒ' ],
+		
+		// Vowels
+		// Vowels without consontant initial must have ᄋ prepended
+		// [^ᄀ-ᄒ]|^ matches the start character or anything but an initial consonant
+		[ '([^ᄀ-ᄒ]|^)wa', '$1와' ],
+		[ '([^ᄀ-ᄒ]|^)wo', '$1워' ],
+		[ '([^ᄀ-ᄒ]|^)we', '$1웨' ],
+		[ '([^ᄀ-ᄒ]|^)wi', '$1위' ],
+		[ '([^ᄀ-ᄒ]|^)ya', '$1야' ],
+		[ '([^ᄀ-ᄒ]|^)ye', '$1예' ],
+		[ '([^ᄀ-ᄒ]|^)yo', '$1요' ],
+		[ '([^ᄀ-ᄒ]|^)yu', '$1유' ],
+		// 'y' diphthongs
+		[ 'ya', 'ᅣ' ],
+		[ 'ᅣe', 'ᅤ' ],
+		[ 'ᅨo', 'ᅧ' ],
+		[ 'ye', 'ᅨ' ],
+		[ 'yo', 'ᅭ' ],
+		[ 'yu', 'ᅲ' ],
+		// 'w' diphthongs
+		[ 'wa', 'ᅪ' ],
+		[ 'ᅪe', 'ᅫ' ],
+		[ 'wo', 'ᅯ' ],
+		[ 'we', 'ᅰ' ],
+		[ 'wi', 'ᅱ' ],
+		// Other diphthongs
+		[ 'ᅩe', 'ᅬ' ],
+		[ 'ᅦu', 'ᅳ' ],
+		[ 'ᅮi', 'ᅴ' ],
+		[ 'ᅦo', 'ᅥ' ],
+		[ 'ᅡe', 'ᅢ' ],
+		[ '([^ᄀ-ᄒ]|^)i', '$1이' ],
+		[ '([^ᄀ-ᄒ]|^)a', '$1아' ],
+		[ '([^ᄀ-ᄒ]|^)u', '$1우' ],
+		[ '([^ᄀ-ᄒ]|^)o', '$1오' ],
+		[ '([^ᄀ-ᄒ]|^)e', '$1에' ],
+		[ 'i', 'ᅵ' ],
+		[ 'a', 'ᅡ' ],
+		[ 'u', 'ᅮ' ],
+		[ 'o', 'ᅩ' ],
+		[ 'e', 'ᅦ' ],
+	];
+
+	var koreanRR = {
+		id: 'kor-rr',
```

Comment:
We usually give them an identified that is based on the shortest language code, so it should be "ko" and not "kor".